include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ * buffer remain associated with the TCP endpoint.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint to abort.
+ *
+ * @retval OT_ERROR_NONE    Successfully aborted the TCP endpoint's connection.
+ * @retval OT_ERROR_FAILED  Failed to abort the TCP endpoint's connection.
+ */
+otError otTcpAbort(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+/**
+ * Deinitializes this TCP endpoint. This means that OpenThread no longer keeps
+ * track of this TCP endpoint and deallocates all resources it has internally
+ * allocated for this TCP endpoint. The application can reuse the memory
+ * backing the TCP endpoint as it sees fit.
+ *
+ * If it corresponds to a live TCP connection, the connection is terminated
+ * unceremoniously (as in @p otTcpAbort). All resources the application has
+ * provided for this TCP endpoint (linked buffers for the send buffer, memory
+ * for the receive buffer, the @p Endpoint structure itself, etc.) are
+ * immediately returned to the application.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure to deinitialize.
+ *
+ * @retval OT_ERROR_NONE    Successfully deinitialized the TCP endpoint.
+ * @retval OT_ERROR_FAILED  Failed to deinitialize the TCP endpoint.
+ */
+otError otTcpEndpointDeinitialize(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+struct otTcpListener;
+typedef struct otTcpListener otTcpListener;
+
+#define OT_TCP_DEFER  ((otTcpEndpoint *) NULL)
+#define OT_TCP_REFUSE ((otTcpEndpoint *) -1)
+
+/**
+ * This callback indicates that an incoming connection that matches this
+ * TCP listener has arrived. The application should provide (via the callback's
+ * return value) a pointer to the @p otTcpEndpoint into which to accept the
+ * incoming connection. The returned @p otTcpEndpoint must already be
+ * initialized (i.e., @p otTcpEndpointInitialize must have been called on it).
+ *
+ * The application may instead choose to return one of two special values.
+ * If the application returns OT_TCP_DEFER, then OpenThread silently ignores
+ * the connection establishment request; the callback will be called again the
+ * next time that the connection establishment request is submitted. This is
+ * valuable if resources are not available to accept the connection, but they
+ * may be available at the time of a future attempt. If the application returns
+ * OT_TCP_REFUSE, then OpenThread sends a "connection refused" message to the
+ * host that attempted to establish a connection.
+ */
+typedef otTcpEndpoint *(*otTcpAcceptReady)(void *aContext, otTcpListener *aListener, const otSockAddr *aPeer);
+
+/**
+ * This callback indicates that the TCP connection is now ready for two-way
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in separate block", can you please check that what I've done is acceptable?